python/nix_csi/subprocessing.py

```
@@ -60,24 +68,38 @@ async def run_console(*args, log_level: int = logging.DEBUG):
+        await asyncio.wait_for(
+            asyncio.gather(
+                stream_output(proc.stdout, stdout_data),
+                stream_output(proc.stderr, stderr_data),
+                proc.wait(),
+            ),
+            timeout=timeout
+        )
+    except asyncio.TimeoutError:
+        proc.kill()
+        await proc.wait()
+        raise GRPCError(Status.DEADLINE_EXCEEDED, f"Command timed out after {timeout}s: {_format_command_preview(args)}")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)